### DISCUSSION TITLE
There are some issues in the example MutatingSootClass.

### DISCUSSION THREAD

**Discussion by SeeUMonster** [soot-oss/SootUp#702] [Q&A]
I'm new to SootUp. So I rebuilt a project, imported SootUp, and wrote a program based on the example code. I have basically copied all the sample code for MutatingSootClass, but when I ran it, I found an error at the first assertEquals, which means that the program did not successfully replace the method. I tried to print all the methods in the replaced class and found that two main methods appeared. That is to say, the program was not successfully replaced and the previous old method was retained. So I debugged and tracked the replacement method and found that:
```
public class CollectionUtils {
  /** Removes the oldValue from the set and adds the newValue afterwards. */
  public static <T> void replace(Set<T> set, T oldValue, T newValue) {
    set.remove(oldValue);
    set.add(newValue);
  }
}
```
The remove method calculates the hash value of the object. The method for calculating hash values using SootMethod is as follows:
```
@Override
  public int hashCode() {
    return Objects.hash(
        getBodySource(), getBodySource().getSignature(), getModifiers(), getParameterTypes());
  }
```
After debugging, I found that the first two parameters have little impact, but the last two parameters, although the contents of Set and List are the same, they are different objects, which can result in different hash values, so that remove will not remove the previous method.

I am not very familiar with the Java language. I was wondering if I could have some feedbacks.

**Comment by swissiety** [comment]
Sry for the late reply - this sounds serious - can you share your test code?
